### PR TITLE
Update virtual floor to better indicate blocked areas.

### PR DESCRIPTION
**Pull request by ZehMatt**
I thought I give #7713 a shot so this is my attempt:

2018-07-29_16-42-39503dacfd70c52bd49

I feel like showing red tiles better indicates that you can not place anything there.

### CODE REVIEW

**Comment by AaronVanGeffen** [comment]
I appreciate you've updated the tool to account for tile ownership! Not much of a fan of the red tiles, though, to be honest.

**Comment by ZehMatt** [comment]
I can undo that part if you want, this felt like the right thing to do.

**Comment by Patrik356b** [comment]
I don't mind the red color, but I also don't like the blue color...

So for those few that doesn't like it, perhaps we could change the color by editing a value in the openrct2 config file? (or change it in the GUI theme editor)

**Comment by ocalhoun6** [comment]
I actually really like the look of the red tiles...
To each his own, though. I supposed there are already a few different user-selectable modes available for the virtual floor -- why not another one?

**Comment by ZehMatt** [comment]
Well I need something more concrete to go on, so what shall the final decision be?

**Comment by AaronVanGeffen** [comment]
I'd be happy to merge it without the red colour. Adding the red colour can then be re-proposed in another PR.

**Comment by Broxzier** [comment]
> (And speaking of the blue ... it's kind of weird that it's not always blue. If you load a map with one of the weird water colors (orange or green), you'll get an orange or green virtual floor.)

That's somewhat expected, and can be confusing indeed. Any colour could be changed with a custom pallette, though I think picking normal blue instead of water-colour works better for most.

**Comment by ZehMatt** [comment]
2018-08-03_18-46-58fc3071afae27da206

Is this acceptable?

**Comment by AaronVanGeffen** [comment]
This completely hides the floor if it is accidentally used completely outside of park boundaries. I would prefer for it to always show the square grid, just transparent when placement is not possible.

I can see why you went with the red colour now, though. We should really look into making this themeable…

**Comment by AaronVanGeffen** [comment]
I think I prefer this to not showing the grid at all… Although I'd like to make the borders thinner — but that would require new sprites, I'm afraid.

screenshot_20180803_210551

**Comment by Broxzier** [comment]
I agree we should make it themable, but since it's not part of the user interface but of the in-game world, it may be better to put the options somewhere else, possibly a new window (this may also be a better place to put https://github.com/OpenRCT2/OpenRCT2/pull/7349).

As for the border, for now I think going with transparent and showing the white border at the park's edges would work best.

**Comment by Broxzier** [comment]
Something like this:

Image

**Comment by ocalhoun6** [comment]
Does this work for construction rights as well?

**Comment by ZehMatt** [comment]
How about this:

2018-08-04_18-30-14ae4127eb37421d7f8

**Comment by Broxzier** [comment]
Not really a fan of the white squares outside of the map, it'd be nicer if the border is only drawn between owned and non-owned land tiles, and that the blue floor is transparent there, like with other places where you cannot build. Also does it take sandbox mode into account?

**Comment by ZehMatt** [comment]
So if I understood you correctly then this should be the desired result:

2018-08-07_00-08-517e62a784082b81572

**Comment by Broxzier** [comment]
Yep, that's perfect!

**Comment by ZehMatt** [comment]
Cool, just pushed my changes. I think I'll leave it at that for now, not sure what you guys are up to with themes.

File: src/openrct2/paint/VirtualFloor.cpp [hunk 1]
```diff
@@ -347,30 +362,30 @@ void virtual_floor_paint(paint_session* session)
     // Edges which are internal to objects (i.e., the tile on both sides
     //  is occupied/lit) are not rendered to provide visual clarity.
     uint8_t dullEdges = 0xF & ~occupiedEdges & ~litEdges;
-    uint8_t paintEdges = (weAreOccupied || weAreLit) ? ~dullEdges : 0xF;
+    uint8_t paintEdges = ((weAreOccupied || weAreLit) && weAreOwned) ? ~dullEdges : 0xF;
 
-    if (paintEdges & 0x1)
+    if ((paintEdges & 0x1))
```
Comment (Broxzier) [inline src/openrct2/paint/VirtualFloor.cpp]: These extra parentheses aren't necessary, and please replace the hardcoded numbers with `EDGE_NE`, `EDGE_SE`, `EDGE_SW` and `EDGE_NW`.

**Pull request merged**